unit_testing/cli/test_cli_sct_crop_image.py

```diff
@@ -1,14 +1,21 @@
-from pytest_console_scripts import script_runner
 import pytest
 import logging
 
+from spinalcordtoolbox.image import Image
+from spinalcordtoolbox.scripts import sct_crop_image
+
 logger = logging.getLogger(__name__)
 
 
-@pytest.mark.script_launch_mode('subprocess')
-def test_sct_crop_image_backwards_compat(script_runner):
-    ret = script_runner.run('sct_testing', '--function', 'sct_crop_image')
-    logger.debug(f"{ret.stdout}")
-    logger.debug(f"{ret.stderr}")
-    assert ret.success
-    assert ret.stderr == ''
+@pytest.mark.sct_testing
+@pytest.mark.usefixtures("run_in_sct_testing_data_dir")
+@pytest.mark.parametrize("path_in,path_out,remaining_args,expected_dim", [
+    ('t2/t2.nii.gz', 't2_crop_xyz.nii', ['-xmin', '1', '-xmax', '-3', '-ymin', '2', '-ymax', '10'], (57, 9, 52)),
+    ('t2/t2.nii.gz', 't2_crop_mask.nii', ['-m', 't2/t2_seg-manual.nii.gz'], (11, 55, 13)),
+    ('t2/t2.nii.gz', 't2_crop_ref.nii', ['-ref', 'mt/mt0.nii.gz'], (37, 55, 34)),
+])
+def test_sct_crop_image_output_has_expected_dimensions(path_in, path_out, remaining_args, expected_dim):
+    """Run the CLI script and verify cropped image has the expected dimensions."""
+    sct_crop_image.main(argv=['-i', path_in, '-o', path_out] + remaining_args)
+    nx, ny, nz, _, _, _, _, _ = Image(path_out).dim
```

Comment:
Aha! Yes, they're necessary to define the relation between voxel coordinates to real world coordinates. Here's a breakdown:

* `nx, ny, nz`: Image dimensions in voxels.
* `nt`: Time dimension (relevant for sequences of scans, esp. dMRI and fMRI).
* `px, py, pz`: The conversion factors between voxels and millimetres (otherwise known as `pixdim` in [`nibabel` docs](https://nipy.org/nibabel/nibabel_images.html#the-image-header)).
* `pt`: The time between scans in milliseconds. 

What I can do is test the remaining 5 values to ensure that `sct_crop_image` doesn't inadvertently change them, since they should be unaffected by cropping.

Reply:
Done in 362455f.